### PR TITLE
Feature/multilanguage caption

### PR DESCRIPTION
**Pull request by oscarssanchez**
### Description of the Change

This PR adds the multi language caption support.

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: includes/admin/api/class-bc-admin-media-api.php [hunk 1]
```diff
@@ -881,6 +883,31 @@ public function ajax_thumb_upload( $account_hash, $video_id, $url, $width, $heig
...
+	 *
+	 * @param string       $account_hash
+	 * @param int          $video_id
+	 */
+	public function ajax_caption_delete( $account_hash, $video_id ) {
+		global $bc_accounts;
+
+		// Set up the account to which we're pushing data
+		$account = $bc_accounts->set_current_account( $account_hash );
+		if ( false === $account ) {
```
Comment (felipeelia) [inline includes/admin/api/class-bc-admin-media-api.php:899]: @oscarssanchez any reason to not use just `if ( ! $account )` here?

File: includes/admin/api/class-bc-admin-media-api.php [hunk 2]
```diff
@@ -903,36 +930,42 @@ public function ajax_caption_upload( $account_hash, $video_id, $raw_captions ) {
...
 
-
 			$url  = esc_url( $caption['source'] );
 			$lang = sanitize_text_field( $caption['language'] );
 			if ( empty( $url ) || empty( $lang ) ) {
 				continue; // Attachment has no URL, fail
 			}
-			$label = isset( $caption['label'] ) ? sanitize_text_field( $caption['label'] ) : '';
+			$label   = isset( $caption['label'] ) ? sanitize_text_field( $caption['label'] ) : '';
+			$default = ( isset( $caption['default'] ) && 'checked' === $caption['default'] ) ? true : false;
```
Comment (felipeelia) [inline includes/admin/api/class-bc-admin-media-api.php:947]: Could we reduce this to just `$default = ( isset( $caption['default'] ) && 'checked' === $caption['default'] )`? Tests will return a boolean anyway.

File: includes/api/class-bc-text-track.php [hunk 3]
```diff
@@ -47,21 +52,40 @@ public function __construct( $url, $language = 'en-US', $kind = 'captions', $lab
...
+			$data['label'] = $this->label;
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Return an array representation of a Text Track for use in API PATCH requests
+	 */
+	public function toArrayPatch() {
```
Comment (felipeelia) [inline includes/api/class-bc-text-track.php:82]: Could we add the @return to the docs?

**Comment by oscarssanchez** [comment]
Thanks @felipeelia  I have addressed your comments on the latest commit, can you take a peek again please?

**Review by felipeelia — DISMISSED** [review]

**Review by felipeelia — APPROVED** [review]

**Pull request merged**